### ISSUE TITLE
Improvements to JSON format of web module

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#437]
* different APIs of the Directions API should use same time unit: seconds (currently routing API has ms)
 * with alternative paths?
 * different layout for via-points? 'legs'
 * turn_angle is in radian

**Comment by michaz** [comment]
> use JSON request instead URL parameter

Are there any examples where this is considered a good idea? I thought the whole point about REST was that a resource is identified by its URL alone, and a shortest path is _the_ perfect example of a resource.

navitia.io seems to agree.

**Comment by karussell** [comment]
> Are there any examples where this is considered a good idea?

ElasticSearch, our Route Optimization API, mapzen, ...

> I thought the whole point about REST was that a resource is identified by its URL alone, and a shortest path is the perfect example of a resource.

Why? There are lots of possible parameters. E.g. it is time, vehicle size etc dependent. I do not see it as a resource. I see it more as a 'calculation job'...

**Comment by michaz** [comment]
> Why?

Because it doesn't change the world, because it returns the same result until it is explicitly changed by something, because it can be cached, ... 

Google search is GET

**Comment by michaz** [comment]
> remove insecure JSONP?

+1

Not only because security, but also because we are not a web framework.

I think in a few years we will say "Oh, you want our API a bit different? Go use a web tool to click together a proxy/gateway which changes it like you want it."

**Comment by karussell** [comment]
> because it returns the same result until it (the road network) is explicitly changed by something, because it can be cached

I would argue against this (as real time road changes will invalidate this and it is unclear which routes to invalidate), but I think this is indeed discussible :)

Let me give you one "beauty" argument as why we shouldn't care much about principled web stuff here:

we'll be able to configure a CH vehicle profiles with a JSON file. Also the default vehicle profiles for non-CH should be configurable by JSON and now, we are forced due to beauty reasons ;), to use the **same JSON** to request our API to dynamically change the vehicle profile per request.

> And, of course, as soon as you start thinking about subscriptions to routes

Yes, but then you could also argue about a resource that you subscribe to ... again: I wouldn't care much. BTW: [navitia api](https://github.com/CanalTP/navitia/wiki/OpenTripPlanner-and-Navitia-comparison#history-1) is reeaally old ;)

**Comment by michaz** [comment]
> we'll be able to configure a CH vehicle profiles with a JSON file. Also the default vehicle profiles for non-CH should be configurable by JSON and now, we are forced due to beauty reasons ;), to use the same JSON to request our API to dynamically change the vehicle profile per request.

I can't follow. Not even which of these is supposed to be good or bad.

So you _want_ to use the same JSON description of a vehicle profile, no matter if you are

- creating a pre-computed profile 
- giving parameters to the flexible router

Correct?

**Comment by michaz** [comment]
Or did you mean that we do not want this, because it would be overdoing it? But then I don't see the connection to POST with JSON, because then we can use JSON for configuring the pre-computation  and URL parameters for the request and nobody cares.

**Comment by karussell** [comment]
> So you want to use the same JSON description of a vehicle profile, no matter if you are

Yes, I meant this and I think this as a good thing :) There is also a third use case where you change defaults e.g. of car to toll=no or ferry=no etc.
